### PR TITLE
Copy & Overwrite Fixes

### LINKED ISSUE #42: Overwrite do not remove leftover data
When a file is copy-overwritten with smaller file, the leftover data is not deleted.
For example:
Original file:
```
aaa
aaa
aaa
```
Copy and overwrite with file:
```
bbb
bbb
```
The result is:
```
bbb
bbb
aaa
```
Instead of expected:
```
bbb
bbb
```

### PR DESCRIPTION
**Pull request by ocerman**
- Fixed overwriting file with a smaller one, FIX #42 
- Handled overwriting directory/symlink with regular file

### CODE REVIEW

**Comment by humdingerb** [comment]
This works nicely!
It would be even better if there was an "Overwrite all" button only if there were more than one file to be copied/moved. Possible?

**Comment by ocerman** [comment]
@humdingerb yes, good idea. Implemented in the last commit.

File: src/GenesisCopyWindow.h [hunk 1]
```diff
@@ -76,6 +78,9 @@ class GenesisCopyWindow : public BWindow
 		int32 GetFirstSelection(void);
 		bool IsDirReadOnly(const char *destination);
 		bool IsRecursiveCopy(const char *source, const char *destination);
+
+		enum ALERT_SKIP_OPTS CopySkipAlert(const char* text);
```
Comment (KapiX) [inline src/GenesisCopyWindow.h]: `enum` keyword is not needed here.

File: src/GenesisCopyWindow.cpp [hunk 2]
```diff
@@ -965,3 +958,48 @@ bool GenesisCopyWindow::IsRecursiveCopy(const char *source, const char *destinat
...
+			return A_SKIP_ABORT;
+		case 1:
+			return m_PossiblyMultipleFiles ? A_SKIP_ALL: A_SKIP_1;
+		default:
+			return A_SKIP_1;
+	}
+}
+
+////////////////////////////////////////////////////////////////////////
+enum ALERT_OVERWR_OPTS GenesisCopyWindow::CopyOverwriteAlert(const char* text)
```
Comment (KapiX) [inline src/GenesisCopyWindow.cpp]: `enum` keyword is not needed here.

**Comment by humdingerb** [comment]
Thanks!

**Pull request merged**